_sass/base/_variables.scss

```diff
@@ -1,63 +1,141 @@
-// Typography
+/* TYPOGRAPHY */
 $font-family-main: 'Source Sans Pro', Helvetica, Arial, sans-serif;
 $font-family-headings: 'Source Sans Pro', Helvetica, Arial, sans-serif;
 $font-family-logo: 'Source Sans Pro', Helvetica, Arial, sans-serif;
 $font-size: 1.25em;
 
-// Padding
+$monospace: Consolas, Menlo, Monaco, Lucida Console, Liberation Mono, DejaVu Sans Mono, Bitstream Vera Sans Mono, Courier New, monospace, sans-serif !default;
+$font-size-code: 0.85em !default;
+$font-height-code: 1.3em !default;
+$border-radius: 4px !default;
+
+/* PADDING */
 $padding-x-large: 23%;
 $padding-large: 9%;
 $padding-medium: 6%;
 $padding-small: 4%;
 $padding-x-small: 3%;
-
-// Sepcific Padding
 $title-padding: 1%;
 $feature-image-padding: 10%;
 
-// Mobile display
+/* MOBILE */
 $xl-break: 2000px;
 $l-break: 1400px;
 $m-break: 1000px;
 $break: 768px;
 $sm-break: 576px;
 $x-sm-break: 300px;
 
-// Brand colours
-$theme-color: #337ab7;
-$brand-color: #fff;
-$background-color: #fff;
-$border-color: rgba(0, 0, 0, 0.1); // rgba recommended if using feature images
-
-// Typography colours
-$text-color: #262626;
-$meta: #595959; //lowest gray for accessible color
-$link-color: $theme-color;
-$selection-color: #D4D4D4; // visible when highlighting text
-$blockquote-color: #EEF7FA; // $background-color;
+/* COLOR SCHEMES */
+/* Load by default dark scheme in systems supporting 'prefers-colors-scheme: dark'
+and load light theme otherwise. Also support switching through schemes by a button. */
+html, html[data-theme="light"] {
+    --gray0:   #ffffff; //
+    --gray1:   #3f3f3f; //
+    --gray2:   #373e47; //
+    --gray3:   #444c56; //
+    --gray4:   #d4d4d4; //
+    --gray5:   #595959; //
+    --gray6:   #383838; //
+    --gray7:   #262626; //
+    --gray8:   #ffffff; //
+    --gray9:   #ffffff; //
+    --red:     #d40059; //
+    --orange:  #3f3f3f; //
+    --yellow:  #d9b600; //
+    --green:   #6bac00; //
+    --blue:    #337ab7; //
+    --magenta: #b37eff; //
+    --cyan:    #00af9e; //
+    --pink:    #ff66c2; //
+    --border:  rgba(0, 0, 0, 0.1); //
+}
```

Comment:
So instead of having `text: $color` you set the variable using `--gray0: #ffffff;`

Instead of using the color as variable name like `--gray0`, I think it would be better if it would describe what it is, like "link", "text" or "h1-color".
Because if you use a different theme it might differ.

I am not too sure how this html tagging work in the scss, so maybe there are restraint on the name.

Reply:
There is a tradeoff, a single color generally have multiple uses so that would lead to:

- confusing variable names (eg. $header-link being used as search color)
- or great increase in the number of variables (eg. $theme-color, $link-color, $link-shadow and $header-bg instead of just $blue)

What I did is comment where each color is being used.

In a sense gray0 to 9 is a gradient between white and black in the light theme and black to white in the dark theme but there are a few ad-hoc fixes.

***

There are two ways of doing dark-mode currently:
- via the JS toggle;
- via `prefers-colors-scheme:` which will use one or the other following the user environment;

Both of them are implemented, that's why there are 4 variable arrays. It will follow 'prefers-color-scheme' and show a dark theme if its enabled in the environment but in both cases will allow the user to toggle the theme via the button.

Reply:
Indeed maybe there was too much variable name, I like the simplification, I'll cook something up to fit both world :)